config_option.go

```diff
@@ -0,0 +1,147 @@
+package gorm
+
+import (
+	"sync"
+	"time"
+
+	"gorm.io/gorm/clause"
+	"gorm.io/gorm/logger"
+	"gorm.io/gorm/schema"
+)
+
+// ConfigOption use functional option for gorm Config.
+type ConfigOption func(c *Config)
```

Comment:
Does this need to be a public type? Maybe name it `configOption` instead?

Reply:
Why should ConfigOption be set as a public type, because there are indeed some scenarios where users need to customize Config option, and at the same time, in order to better expand and upgrade gorm v2, so here is implemented using functional option, you can look at the Apply method This is the intention I want to express to facilitate the upgrade later.